04-onion-routing.md

```diff
@@ -601,6 +606,20 @@ If the `cltv-expiry` is too low, the final node MUST fail the HTLC:
 
 1. type: 17 (`final_expiry_too_soon`)
 
+If the `outgoing_cltv_value` does not match the `ctlv-expiry` of the
+HTLC at the final hop:
+
+1. type: 18 (`final_incorrect_cltv_expiry`)
+2. data:
+   * [4:cltv-expiry]
+
+If the `amt_to_forward` does not match the `incoming_htlc_amt` of
```

Comment:
That doesn't work if we allow non-precise payments, though.  The second-to-final node can take more fee than expected, and the last node won't detect it if it allows anything other than a specific amount  (eg. donation address, and even before #139 overpaying detection was a SHOULD not a MUST)...